pilot/cmd/pilot-discovery/main.go

```diff
@@ -109,7 +109,8 @@ func init() {
 	discoveryCmd.PersistentFlags().StringSliceVar(&serverArgs.MCPServerAddrs, "mcpServerAddrs", []string{},
 		"comma separated list of MCP server addresses with "+
 			"mcp:// (insecure) or mcps:// (secure) schema, e.g. mcps://istio-galley.istio-system.svc:9901")
-	serverArgs.MCPCredentialOptions.AttachCobraFlags(discoveryCmd)
+	discoveryCmd.PersistentFlags().MarkDeprecated("mcpServerAddrs", "Use --meshConfig instead, and specify in MeshConfig.ConfigSources[].Address")
```

Comment:
There is a potential chicken-and-egg problem with Pilot getting the MCP server address from MeshConfig if the MeshConfig itself is ever delivered to Pilot via MCP. If loading from a local file is desired, it might be good to rename MeshConfig to PilotComponentConfig and cleanup the use of flags and envvars to be more consistent. 

cc @ozevren

Reply:
The solution is to not deliver mesh config via Galley. There are somethings that are best obtained locally. Otherwise, we create an unnecessary dependency on Galley to even get the system started.

Reply:
Yes, currently the mesh config is get first from local file, otherwise from configmap.

Reply:
Isn't it possible to have a local config tailored for each component (i.e. Pilot) that contains only enough information to establish initial connectivity to a config source (i.e. Galley)?

Reply:
That local config is MeshConfig.

Reply:
Related: [Kubernetes Component configuration](https://goo.gl/opSc2o). The k8s/CRD specific parts aren't as relevant, but I think the general idea still applies. 

A local file is reasonable but we ought to be careful how much we put in there (vs. adding user facing APIs via CRD). I think we're headed in that direction already, with most of the feature-level fields in MeshConfig be deprecated. Given that @rshriram is suggesting MeshConfig is really the pilot's local configuration, would it make sense to create a new PilotComponentConfig which is Pilot specific to contain these types of options that a mesh admin would need to tweak. Service owners shouldn't need to touch this new PilotComponentConfig and would use the feature-level APIs (e.g. per-service tracing CRD).